### ISSUE TITLE
Upgrade to 5.0.0dev API for making workflows

### ISSUE THREAD

**Issue by LichMaster98** [isi-vista/vista-pegasus-wrapper#48]
> we did do 5.0beta1 release that folks should consider moving it. it has a new workflows api . Migration Guide: https://pegasus.isi.edu/docs/5.0.0dev/migration.html . Release notes: https://pegasus.isi.edu/2020/07/27/pegasus-5-0-beta-released/

**Comment by spigo900** [comment]
Some notes on the migration guide -- what's relevant as far as the wrapper goes:

- Do any users of this wrapper use Python 3.4 or below? If so then we can't migrate yet: "The Pegasus.api package requires Python3.5+."
- The sites catalog changed from XML to YAML
  - There's a [conversion tool](https://pegasus.isi.edu/docs/5.0.0dev/manpages/pegasus-sc-converter.html) we can use to update ours
- The DAX API changed
  - The new version is under `Pegasus.api`
  - `ADAG` was renamed to `Workflow`
  - `File`s changed
    - Have to add them through the replica catalog, not the graph
    - Can create the replica catalog programmatically
  - `Executable`s changed
    - Now referred to as "transformations"
    - Have to add them through the transformation catalog, not the graph
    - Can create the transformation catalog programmatically
  - `Job` had its API changed slightly
    - Basically the same
    - Replaces `uses()` with `add_inputs()` and `add_outputs()`
- There's now a programmatic way to build a properties file
  - I think the properties file is/does the same things as `pegasus.conf`

**Comment by spigo900** [comment]
@marjorief Re: an effort estimate, just based on the migration guide notes, I think the immediate code changes are more like 2 days (or 3) than 2 months.

However, because it's a major version change and a new API, I suspect there might be other changes to how things work not covered by these notes or the guide. I estimate those things might make this more like 2 weeks (as a conservative estimate) to make the migration guide changes + test to confirm things still work and fix the things that don't. (Assuming we already have Pegasus 5.0.0dev set up somewhere.)

**Comment by marjorief** [comment]
> (Assuming we already have Pegasus 5.0.0dev set up somewhere.)
Is the first step setting Pegasus5.0.5dev on some submit VM such that we can test it? 

I am going to put this in the "think about in Jan. bucket" unless @danielnapierski wants to push on getting us to the point where we are actually testing Pegasus5.0.5dev.

**Comment by LichMaster98** [comment]
@marjorief We do happen to accidentally have Pegasus 5 (unsure the exact version number) configured on `saga03` as it was accidentally upgraded while none of the submit nodes were.

**Comment by danielnapierski** [comment]
Newer link to migration notes: https://pegasus.isi.edu/docs/5.0.0/user-guide/migration.html#migration-notes

Regarding Python3.5 requirement.  Should a message go out to `saga-users`?  @marjorief

**Comment by marjorief** [comment]
@danielnapierski -- is the Python 3.5 requirement for all saga users, or just for saga users who are using pegasus?

**Comment by danielnapierski** [comment]
@marjorief Only Pegasus users I believe.
